Stack.yml

```diff
@@ -0,0 +1,132 @@
+    AllowedPattern: (?!-)[a-zA-Z0-9-.]{1,63}(?<!-)
+    ConstraintDescription: must be a valid DNS zone name.
+  AcmCertificateArn:
+    Type: String
+
+Resources:
+  Bucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Ref 'BucketName'
+      AccessControl: PublicRead
+      WebsiteConfiguration:
+        IndexDocument: index.html
+        ErrorDocument: error/404/index.html
+        RoutingRules:
+          - RoutingRuleCondition:
+              HttpErrorCodeReturnedEquals: '404'
+              KeyPrefixEquals: /
+            RedirectRule:
+              HostName: !Sub
+                - '${SubDomainName}.${HostedZoneName}.'
+                - SubDomainName: !Ref SubDomainName
+                  HostedZoneName: !Ref HostedZoneName
+              ReplaceKeyPrefixWith: error/404/
+  WebsiteCloudFront:
+    Type: AWS::CloudFront::Distribution
+    Properties:
+      DistributionConfig:
+        Enabled: true
+        Comment: CloudFront Distribution pointing to S3 Bucket
+        Origins: 
+        - DomainName: !Join [ '', !Split [ 'http://', !GetAtt 'Bucket.WebsiteURL' ] ]
+          Id: S3Origin
+          CustomOriginConfig:
+            OriginProtocolPolicy: 'http-only'
+        HttpVersion: 'http2'
+        Aliases: 
+          - !Sub 
+            - '${SubDomainName}.${HostedZoneName}'
+            - SubDomainName: !Ref SubDomainName
+              HostedZoneName: !Ref HostedZoneName 
+        CustomErrorResponses:
+          - ErrorCode: 403
+            ResponseCode: 404
+            ResponsePagePath: '/error/404/index.html'
+        DefaultCacheBehavior:
+          TargetOriginId: S3Origin
+          ViewerProtocolPolicy: redirect-to-https
+          AllowedMethods:
+            - GET
+            - HEAD
+            - OPTIONS
+          Compress: true
+          ForwardedValues:
+            QueryString: false
+            Cookies:
+              Forward: none
+          ResponseHeadersPolicyId: !Ref WebsiteCloudFrontPolicy
+        PriceClass: PriceClass_All
+        ViewerCertificate:
+          AcmCertificateArn: !Ref 'AcmCertificateArn'
+          SslSupportMethod: sni-only
+  WebsiteCloudFrontPolicy:
+    Type: AWS::CloudFront::ResponseHeadersPolicy
+    Properties:
+      ResponseHeadersPolicyConfig:
+        Name: API-Docs-Security-Header-Policy
+        Comment: Vapor API Docs security header policy
+        CustomHeadersConfig:
+          Items:
+            - Header: Permission-Policy
+              Override: false
+              Value: accelerometer=(), camera=(), geolocation=(), gyroscope=(), magnetometer=(), microphone=(), payment=(), usb=(), interest-cohort=()
+            - Header: server
+              Override: true
+              Value: 'Vapor API Docs'
+        SecurityHeadersConfig:
+          ContentSecurityPolicy:
+            ContentSecurityPolicy: default-src 'none'; script-src 'self'; img-src 'self'; style-src 'self'; font-src 'self'
+            Override: false
```

Comment:
Lets comment this out for now until we work out how the DocC stuff works otherwise it could break the site